README.md

```diff
@@ -58,124 +95,177 @@ plugins/gtm-martech/*
+  } else if (el.classList.contains('section')) {
+    // Do something on each section to push to DataLayer
+  }
+}
+```
+
+## API Reference
+
+This plugin exports several functions to manage the marketing libraries:
+
+--- 
+
+### `new GtmMartech(martechConfig)`
+
+Initializes the plugin. This should be called in the `scripts.js` outside any lifecycle phase.
+
+- **`martechConfig`** `{Object}`: Configuration for this plugin.
+  - `analytics` `{Boolean}`: Enable analytics. Default: `true`.
+  - `dataLayer` `{Boolean}`: Enable GTM Data Layer. Default: `true`.
+  - `dataLayerInstanceName` `{String}`: Global name for the GTM Data Layer instance. Default: `'gtmDataLayer'`.
+  - `tags` `{String[]}`: Array of GA4 Measurement Ids to load.
+  - `containers` `{Object|String[]|String}`: Configuration for GTM Containers, or an Array of GTM Container Ids to load during the lazy phase, or a single GTM Container Id to load during the lazy phase.
+    - `lazy` `{String[]}`: Array of GTM Container Ids to load in the lazy phase.
+    - `delayed` `{String[]}`: Array of GTM Container Ids to load in the delayed phase.
+  - `pageMetadata` `{Object}`: A set of key-value pairs to pass to the GA4 tag initializer.
+  - `consent` `{Boolean}`: Enable consent. Default: `true`
+  - `consentCallback` `{Function|undefined}`: A function that will perform consent validation. Returns a promise that resolves to an object, which will be passed to the GA for update.
+  - `decorateCallback` `{Function}`: A function that can decorate HTML elements for events. Passed all sections & blocks found.
+
+---
+
+### `pushToDataLayer(payload)`
+Pushes a generic payload to the Adobe Client Data Layer.
+
+- **`payload`** `{Object}`: The data object to push.
+
+---
+
+### `updateUserConsent(consent)`
+Updates the consent according to the []`gtag.js` implementation](https://developers.google.com/tag-platform/security/guides/consent?consentmode=advanced#implementation_example)
+
+- **`consent`** `{Object}`: An object detailing user consent choices.
+
+---
+
+## An Example Site
 
 An example of this plugin in use can be found on the [AEM GTM Martech demo site](https://main--aem-gtm-martech-site--adobe-rnd.aem.page/).
 
 
 ## FAQ
 
-### The earlies GTM Containers are loaded Lazy. I need the page view data immediately.
-
-While the GTM containers are loaded lazy, the GA4 library is loaded in the eager phase. This loading will send a collection event for a page view. Any custom data passed via the `pageMetadata` configuration property will be passed along with the page view event.
+### I need the page view data immediately, when is it captured?
 
+All GA4 scripts are imported to the page during the Eager Phase. During this process a page view event is dispatched, along with any custom metadata provided to during plugin initialzation. Even if the GTM containers are loaded lazily, this data will be captured as soon as Google's library is conifgured to dispatch it.
```

Comment:
It will only be submitted when the GTM loads though, which means the page view is reported at the end of the lazy as a "bottom of page event" one the page is fully rendered. This is also what we do for the Adobe stack

Reply:
I tested this and didnt load any GTM libraries and it still sends the page view. According to docs, that is supposed to happen using the gtag function.

Reply:
ok, nice! Conceptually, I'm still undecided if this is right though. The page is not rendered until end of "lazy"… so technically it was not viewed in "eager" yet. Might be nitpicking, but to me that's kinda misleading 😅

Reply:
Hm. So technically the view event isn't dispatched until GA decides to send the first batch of events. So maybe:

```suggestion
All GA4 scripts are imported to the page during the Eager Phase. During this process a `page_view` event occurs, which includes any custom metadata provided to during plugin initialization. This `page_view` event will be dispatched, as soon as the Google library decides it should be sent, regardless of when any other phase's GTM libraries are loaded.
```

Thoughts?

Reply:
Approved :)